kernel/src/main/java/org/kframework/compile/AddSortInjections.java

```diff
@@ -303,6 +303,10 @@ private void match(Production prod, Sort declaredSort, Sort actualSort, Map<Sort
                         .collect(Collectors.toSet());
     }
 
+    public Sort topSort(K term) {
+      return sort(term, freshSortParam());
+    }
+
```

Comment:
Why is this called `topSort`?
Please add some comments because it's not clear what top sort means in this context.